app/agents/RecruitmentStageManagerAgent.py

```diff
@@ -0,0 +1,213 @@
+            self.agent.logger.info(
+                f"No recruitment stages found. Recruitment stage with recruitment: {self.agent.recruitment_id} and identifier: {self.agent.identifier} to be created."
+            )
+        else:
+            self.agent.if_created = True
+            self.agent.recruitment_stage = recruitment_stages[0]
+
+            await self.check_stage_status()
+            self.agent.logger.info(
+                f"Found recruitment stage with id: {recruitment_stages[0]._id}. No recruitment stage objects will be created."
+            )
+
+    async def check_stage_status(self):
+        if self.agent.recruitment_stage.status == RecruitmentStageStatus.DONE:
+            self.agent.logger.info("Stage status is DONE. Exiting agent...")
+
+            await self.agent.stop()
+
+
+class PrepareRecruitmentStage(spade.behaviour.OneShotBehaviour):
+    """Creates recruitment stage object (if not present)"""
+
+    agent: RecruitmentStageManagerAgent
+
+    async def run(self):
+        self.agent.logger.info("PrepareRecruitmentStage behaviour run.")
+
+        await self.create_recruitment_stage()
+
+        self.agent.manage_stage_behav = ManageState(period=MANAGE_STAGE_PERIOD)
+        self.agent.add_behaviour(self.agent.manage_stage_behav)
+
+    async def create_recruitment_stage(self):
+        if self.agent.if_created:
+            return
+
+        self.agent.recruitment_stage_attr.update(
+            {"_id": "", "recruitment_id": self.agent.recruitment_id, "result": 0.0}
+        )
+        self.agent.recruitment_stage = RecruitmentStage(
+            **self.agent.recruitment_stage_attr
+        )
+
+        id = self.agent.recruitment_stage_module.create(self.agent.recruitment_stage)
+        self.agent.recruitment_stage._id = id
+
+
+class ManageState(spade.behaviour.PeriodicBehaviour):
+    """Behaviour representing ManageStage protocol (mainly, asking RM agent for the permission to start)"""
+
+    agent: RecruitmentStageManagerAgent
+
+    async def run(self):
+        self.agent.logger.info("ManageStage behaviour run.")
+
+        await self.send_and_receive()
+
+    async def send_and_receive(self):
+        msg = await self.agent.prepare_message(
+            self.agent.rm_jid,
+            "request",
+            "start_request",
+            MessageType.START_REQUEST,
+            [f"{self.agent.jid}", f"{self.agent.recruitment_stage.priority}"],
+        )
+
+        self.agent.logger.info("Sending message to rm agent with the request to start.")
+        await self.send(msg)
+
+        msg = await self.receive(timeout=10)
+        if msg is None:
+            return
+
+        _, data = await self.agent.get_message_type_and_data(msg)
+        start_allowed = await self.evaluate_stage_start(data[0])
+        if start_allowed:
+            self.agent.track_stage_behav = TrackStage(period=TRACK_STAGE_PERIOD)
+            self.agent.add_behaviour(self.agent.track_stage_behav)
+
+            self.kill()
```

Comment:
co to robi? brzmi brutalnie

Reply:
konczy zachowanie

Reply:
w tym kontekscie chce skonczyc ManageState (bo juz dostal zgode od RM na start) i zaczac TrackStage